### DISCUSSION TITLE
Encapsulated custom events not working

### DISCUSSION THREAD

**Discussion by kcap** [alpinejs/alpine#2227] [7. Bugs (archived)] [answered]
I'm trying to encapsulate an custom event in Alpine.data () but it doesn't fire. It only fires if it is in the attribute:

`["@custom-event.window"]() { ... }`
does not work
`<div @custom-event.window = "...">`
works

https://codepen.io/bitard999/pen/LYjpoeW

**Comment by SimoTod** [comment, marked as answer]
You are not using x-bind, you are just declaring a property called `@custom-event.window` in x-data which is different from a directive
